envsec/internal/flow/init.go

```diff
@@ -0,0 +1,199 @@
+package flow
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/manifoldco/promptui"
+	"github.com/samber/lo"
+	"go.jetpack.io/envsec/internal/git"
+	"go.jetpack.io/pkg/api/api"
+	projectsv1alpha1 "go.jetpack.io/pkg/api/gen/priv/projects/v1alpha1"
+	"go.jetpack.io/pkg/auth/session"
+	"go.jetpack.io/pkg/id"
+	"go.jetpack.io/typeid"
+)
+
+// flow:
+// 0. Ask if you want to overwrite existing config [y/N]
+// 1. Link to an existing project? [Y/n]
+// 2a. What project would you like to link to? (sorted by repo/dir match)
+// 2b. What’s the name of your new project?
+
+type Init struct {
+	Client                *api.Client
+	PromptOverwriteConfig bool
+	Token                 *session.Token
+	WorkingDir            string
+}
+
+func (i *Init) Run(ctx context.Context) (id.ProjectID, error) {
+	if i.PromptOverwriteConfig {
+		overwrite, err := i.overwriteConfigPrompt()
+		if err != nil {
+			return id.ProjectID{}, err
+		}
+		if !overwrite {
+			return id.ProjectID{}, errors.New("aborted")
+		}
+	}
+
+	// TODO: printTeamNotice will be a team picker once that is implemented.
+	if err := i.printTeamNotice(ctx); err != nil {
+		return id.ProjectID{}, err
+	}
+	linkToExisting, err := i.linkToExistingPrompt()
+	if err != nil {
+		return id.ProjectID{}, err
+	}
+	if linkToExisting {
+		return i.showExistingListPrompt(ctx)
+	}
+	return i.createNewPrompt(ctx)
+}
+
+func (i *Init) overwriteConfigPrompt() (bool, error) {
```

Comment:
Instead of overwrite, I think you can make this a confirmation prompt for all projects.

If project has never been initialized the prompt is:
`Set up project in "<dir>"? [Y/n]` defaulted to yes.

If the project already exists, then you do:
`Project already exists. Reset project in "<dir>"? [y/N]` defaulted to no